### PR TITLE
Problem: latest pre-release of SDK is not used

### PR DESCRIPTION
**Pull request by tomtau**
Solution:
- bumped dependency to 0.43.0-rc3
- adjusted the code (moved to ibc-go, new modules...)
- module code changes + migration code
- fixed integration tests
- added new tests for the vesting bug

### CODE REVIEW

**Review by yihuang — COMMENTED** [review]
There's this line in cosmos-sdk's simapp: https://github.com/cosmos/cosmos-sdk/blob/release/v0.43.x/simapp/app.go#L438, not sure if it's necessary.

File: cmd/chain-maind/app/app.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 	config.SetConfig()
 	encodingConfig := app.MakeEncodingConfig()
 	initClientCtx := client.Context{}.
-		WithJSONMarshaler(encodingConfig.Marshaler).
+		WithJSONCodec(encodingConfig.Marshaler).
```
Comment (yihuang) [inline cmd/chain-maind/app/app.go:50]:
```suggestion
		WithCodec(encodingConfig.Marshaler).
```

JSONCodec is deprecated here: https://github.com/cosmos/cosmos-sdk/commit/7bca49e2d29460dae7c55fa77aee69ab0810d1f0

File: cmd/chain-maind/app/app.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 }
 
 func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
-	authclient.Codec = encodingConfig.Marshaler
+	// authclient.Codec = encodingConfig.Marshaler
```
Comment (yihuang) [inline cmd/chain-maind/app/app.go:87]: https://github.com/cosmos/cosmos-sdk/commit/d7dd1d7affc02a819845802da540bc2742d82074#diff-146f73b0327c426deb3126e44f9234c7a6fe190c1bcb4c98ab8366b3485922edR81
There also this change in new version, is that relevant?

**Review by saed219 — APPROVED** [review]

File: cmd/chain-maind/app/app.go [hunk 3]
```diff
@@ -46,8 +46,8 @@
~func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 	config.SetConfig()
 	encodingConfig := app.MakeEncodingConfig()
```
[the remark this answers is not in the document]
Comment (saed219) [reply cmd/chain-maind/app/app.go:48]: ****

File: x/nft/client/cli/cli_test.go [hunk 4]
```diff
@@ -29,11 +35,21 @@ type IntegrationTestSuite struct {
 	network *network.Network
 }
 
+func GetApp(val network.Validator) servertypes.Application {
```
Comment (JayT106) [inline x/nft/client/cli/cli_test.go:38]: There are two same`GetApp` in the different tests. perhaps can move it to `testutil/test_helpers.go`

**Review by devashishdxt — APPROVED** [review]

**Review by yihuang — APPROVED** [review]
LGTM

**Pull request merged**